app/helpers/govuk_exit_this_page_link_helper.rb

```diff
@@ -0,0 +1,19 @@
+module GovukExitThisPageLinkHelper
+  def govuk_exit_this_page_link(
+    text: Govuk::Components.config.default_exit_this_page_text,
+    href: Govuk::Components.config.default_exit_this_page_redirect_url,
+    classes: [],
+    **html_attributes,
+    &block
+  )
+    link_classes = Array.wrap(classes).append(%w(govuk-skip-link govuk-js-exit-this-page-skiplink))
+
+    html_attributes_with_data_module = { data: { module: "govuk-skip-link" } }.deep_merge(html_attributes)
+
+    return link_to(href, class: link_classes, **html_attributes_with_data_module, &block) if block_given?
+
+    link_to(text, href, class: link_classes, **html_attributes_with_data_module)
```

Comment:
Would this work - passing both `block` and `text` to `link_to` and relying on [the logic there](https://github.com/rails/rails/blob/main/actionview/lib/action_view/helpers/url_helper.rb#L212-L222) to have the block override the text param?

```suggestion
    link_to(text, href, class: link_classes, **html_attributes_with_data_module, &block)
```

Reply:
Good point, that works just fine.